File: benchmark/2026-09-17-7677581(latest)/semantic-compression/probes/cpp/F17.P2.cpp
#include <cstdint>
#include <cstdio>

std::int32_t released = 0;

std::int32_t cleanup_count() {
// BEGIN PROBE F17.P2
struct Handle {
    std::int32_t id;
    ~Handle() { ++released; }
};
{
    Handle h{1};
}
return released;
// END PROBE F17.P2
}

int main() {
    std::printf("%d\n", cleanup_count());
}

File: benchmark/2026-09-17-7677581(latest)/semantic-compression/probes/go/F17.P2.go
package main

import "fmt"

var released int32

// BEGIN PROBE F17.P2
type Handle struct{ id int32 }

func (h Handle) Close() { released++ }

func lifetime() int32 {
	func() {
		h := Handle{1}
		defer h.Close()
	}()
	return released
}

// END PROBE F17.P2

func main() {
	fmt.Println(lifetime())
}
